src-electron/main-process/modules/wallet-rpc.js

```diff
@@ -808,6 +845,170 @@ export class WalletRPC {
     });
   }
 
+  async updateLocalLNSRecords() {
+    try {
+      const addressData = await this.sendRPC("get_address", { account_index: 0 }, 5000);
+      if (addressData.hasOwnProperty("error") || !addressData.hasOwnProperty("result")) {
+        return;
+      }
+
+      // Pull out all our addresses from the data and make sure they're valid
+      const results = addressData.result.addresses || [];
+      const addresses = results.map(a => a.address).filter(a => !!a);
+      if (addresses.length === 0) return;
+
+      const records = await this.backend.daemon.getLNSRecordsForOwners(addresses);
+
+      // We need to ensure that we decrypt any incoming records that we already have
+      const currentRecords = this.wallet_state.lnsRecords;
+      const recordsToUpdate = new Set(this.purchasedNames);
```

Comment:
This won't work as `this.purchasedNames` is a `Set<String>` where as we want the records to be `Set<{ name, type }>`.

Also because we are using a dictionary, sets wont work